include/PathOrderOptimizer.h

```diff
@@ -636,18 +636,35 @@ class PathOrderOptimizer
             return vert;
         }
 
+        //Precompute segments lengths because we are going to need them multiple times
+        std::vector<coord_t> segments_sizes(path.converted->size());
+        coord_t total_length = 0;
+        for(const auto& [i, here]: **path.converted | ranges::views::enumerate)
+        {
+            const Point &next = (*path.converted)[(i + 1) % path.converted->size()];
+            coord_t segment_size = vSize(next - here);
+            segments_sizes[i] = segment_size;
+            total_length += segment_size;
+        }
+
         size_t best_i;
         float best_score = std::numeric_limits<float>::infinity();
         for (const auto& [i, here] : **path.converted | ranges::views::enumerate)
```

Comment:
Rather then performing the if-check on line below for every iteration, you could also skip the last element using the ranges `drop_last` utility
```suggestion
        for (const auto& [i, here] : **path.converted | ranges::views::enumerate | ranges::views::drop_last(/*is_path_closed()*/true ? 1 : 0))
```

Reply:
This is very nice, I was looking for a clean way to do this but I'm not familiar enough with `std::ranges`